test/handle_access_token_response_test.dart

```diff
@@ -138,6 +144,31 @@ void main() {
       expect(credentials.accessToken, equals('access token'));
     });
 
+    test('with a URL-encoded content-type returns the correct credentials', () {
+      var body = 'token_type=bearer&access_token=access%20token';
+      var credentials = handle(new http.Response(body, 200, headers: {'content-type': 'application/x-www-form-urlencoded'}));
+      expect(credentials.accessToken, equals('access token'));
+      expect(credentials.tokenEndpoint.toString(),
+          equals(tokenEndpoint.toString()));
+    });
+
+    test('with custom getParameters() returns the correct credentials', () {
+      var body = 'token_type=bearer&access_token=access%20token';
+      var credentials = handle(new http.Response(body, 200, headers: {'content-type': 'text/plain'}),
+          getParameters: (String contentType, String body) {
+        return body.split('&').fold<Map<String, dynamic>>({}, (out, str) {
+          var equals = str.lastIndexOf('=');
+          if (equals < 1 || equals >= str.length - 1) return out;
+          var key = str.substring(0, equals);
+          var value = Uri.decodeComponent(str.substring(equals + 1));
+          return out..[key] = value;
+        });
```

Comment:
This is a lot of parsing logic that's not really necessary—the caller doesn't care how this is parsed. Just make the body something simple like a JSON doc that starts with the character `_`.